scripts/updateUserGeoLocation.js

```diff
@@ -0,0 +1,65 @@
+// Script to remove lastIP and add geo location using the IP for existing users
+// For some users and their IPs, the geo location is not found (some are reserved IP addresses) - IP is removed anyways for them from the database (So, they dont have IP as well as geo location)
+// TODO: Set geo location for existing users that we don’t have that data for yet given their current IP address, or move straight to adding a last known geo property set
+
+require('coffee-script');
+require('coffee-script/register');
+global._ = require('lodash')
+_.str = require('underscore.string')
+database = require('../server/commons/database')
+User = require('../server/models/User')
+co = require('co')
+countryList = require('country-list')()
+geoip = require('@basicer/geoip-lite')
+
+database.connect()
+var count = 0
+
+co(function*() {
+
+    //var cursor = User.find({"geo": {$exists: false}, "lastIP": {$exists: true}},{"lastIP": 1, "geo": 1, "name": 1, "email": 1}).cursor()   // to update all users
+    var cursor = User.find({"geo": {$exists: false}, "lastIP": {$exists: true}},{"lastIP": 1, "geo": 1, "name": 1, "email": 1}).limit(1).cursor()   // to update limited number of users
```

Comment:
You may be able to use `toArray()` instead of `cursor()` here for cleaner iteration:
https://docs.mongodb.com/manual/reference/method/cursor.toArray/

Reply:
I am using the mongoose cursor which is a wrapper around mongodb cursor([http://mongoosejs.com/docs/api.html#query_Query-cursor]). It iterates through the documents without loading them all into the memory. It is an alternative to using streams() which processes results that are too big to fit into memory at once. (http://thecodebarbarian.com/cursors-in-mongoose-45)